core/arch/riscv/kernel/boot.c

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2015-2021, Linaro Limited
+ * Copyright 2022 NXP
+ */
+
+#include <assert.h>
+#include <compiler.h>
+#include <config.h>
+#include <console.h>
+#include <keep.h>
+#include <kernel/boot.h>
+#include <kernel/linker.h>
+#include <kernel/misc.h>
+#include <kernel/panic.h>
+#include <kernel/thread.h>
+#include <riscv.h>
+#include <stdio.h>
+#include <trace.h>
+#include <util.h>
+
+#include <platform_config.h>
+
+#define PADDR_INVALID               ULONG_MAX
+
+uint32_t sem_cpu_sync[CFG_TEE_CORE_NB_CORE];
+
+static void init_runtime(unsigned long pageable_part __unused)
+{
+}
+
+void init_sec_mon(unsigned long nsec_entry __maybe_unused)
```

Comment:
Do you need this?

Reply:
Will decide it later, if it to use OpenSBI with security monitor extension or to implement security monitor in OP-TEE.

Reply:
I mean the current implementation we have uses it.  But not sure to keep it in the end, as there are many approches for security monitor. The one we implement now is based on OpenSBI extension, but those are not specifications yet.